components/ruby/ruby-34/Makefile

```diff
@@ -0,0 +1,167 @@
+PKG_MACROS += RUBY_LIB_VER=$(RUBY_LIB_VER)
+PKG_MACROS += RUBY_CPU=$(RUBY_CPU)
+
+GENERATE_EXTRA_CMD += | $(GSED) -e 's|x86_64|$$(RUBY_CPU)|g'
+
+# Apply some of the edits here--including to the sed scripts themselves.
+# These modifications of ruby.1 must occur after ruby
+# is installed, not before.  Although there are some other patches
+# to ruby.1 that occur before, the ruby installation does some
+# formatting of ruby.1 that won't occur if the below changes
+# are applied.
+# Fix generated rbconfig.rb using /usr/bin/sed; gsed doesn't
+# work for some of the commands in rbconfig.sedscript.
+COMPONENT_POST_INSTALL_ACTION += \
+	$(GSED) -e "s/RUBY_VER/$(RUBY_VER)/" \
+	    -e "s/RUBY_LIB_VER/$(RUBY_LIB_VER)/" Solaris/gem.1-generic \
+	    > Solaris/gem.1 ; \
+	$(GSED) -e "s/RUBY_VER/$(RUBY_VER)/" \
+	    -e "s/RUBY_LIB_VER/$(RUBY_LIB_VER)/" Solaris/ruby.1.sedscript \
+	    > Solaris/ruby.1.sedscript.mod ; \
+	$(GSED) -e  "s/RUBY_VER/$(RUBY_VER)/" \
+	    -e "s/RUBY_LIB_VER/$(RUBY_LIB_VER)/" \
+	    -e "s,%%CC%%,$(CC)," \
+	    -e "s,%%CXX%%,$(CXX)," \
+	    Solaris/rbconfig.sedscript \
+	    > Solaris/rbconfig.sedscript.mod ; \
+	$(GSED) -i -f Solaris/ruby.1.sedscript.mod \
+	    $(PROTORUBYDIR)/share/man/man1/ruby.1 ; \
+	/usr/bin/sed -f Solaris/rbconfig.sedscript.mod \
+	$(PROTO_RBCONFIG_FILE) > rbconfig.rb.mod ; \
+	$(MV) rbconfig.rb.mod $(PROTO_RBCONFIG_FILE) ; \
+	$(MW) $(PROTO_DIR)/usr/include $(PROTORUBYDIR) ;
+
+COMPONENT_POST_INSTALL_ACTION += \
+	encdir="$(PROTORUBYDIR)/lib/ruby/$(RUBY_LIB_VER)/$(RUBY_CPU)-solaris2.11/enc"; \
+	filelist=`ls -1I trans $$encdir`; \
+	subdirlist=`ls -1 $$encdir/trans`; \
+        for f in $$filelist; \
+	do \
+          /usr/bin/elfedit -e 'dyn:runpath "$(USRDIR)/$(COMPONENT_NAME)/$(RUBY_VER)/lib/$(MACH64)"' $$encdir/$$f; \
+	done; \
+        for f in $$subdirlist; \
+	do \
+          /usr/bin/elfedit -e 'dyn:runpath "$(USRDIR)/$(COMPONENT_NAME)/$(RUBY_VER)/lib/$(MACH64)"' $$encdir/trans/$$f; \
+	done;
+
+# Set the locale to generate documentation correctly
+COMPONENT_BUILD_ENV += LC_ALL=en_US.UTF-8
+COMPONENT_INSTALL_ENV += LC_ALL=en_US.UTF-8
+
+CONFIGURE_PREFIX =	$(USRDIR)/$(COMPONENT_NAME)/$(RUBY_VER)
+
+CONFIGURE_OPTIONS +=	--with-rubylibprefix=$(CONFIGURE_LIBDIR.32)/ruby
+CONFIGURE_OPTIONS +=	--enable-shared
+CONFIGURE_OPTIONS +=	--enable-rpath
+CONFIGURE_OPTIONS +=	--disable-dtrace
+# does not build
+CONFIGURE_OPTIONS +=    --enable-yjit
+CONFIGURE_OPTIONS +=    --enable-rjit
+CONFIGURE_OPTIONS +=    --enable-debug-env
+CONFIGURE_OPTIONS +=    --with-thread=pthread
+CONFIGURE_OPTIONS +=    --with-rdoc=ri,html
+CONFIGURE_OPTIONS +=    --with-ruby-version=$(RUBY_VER)
+# Don't need docs for ruby C source files
+CONFIGURE_OPTIONS +=	--disable-install-capi
+CONFIGURE_OPTIONS +=	--disable-option-checking
+CONFIGURE_OPTIONS +=	--with-openssl-dir=$(OPENSSL_PREFIX)
+CONFIGURE_OPTIONS +=	CFLAGS="$(CFLAGS)"
+
+CLEAN_PATHS +=		Solaris/gem.1 Solaris/rbconfig.sedscript.mod \
+			Solaris/ruby.1.sedscript.mod
+
```

Comment:
Please add
COMPONENT_INSTALL_ENV += HOME=/tmp
otherwise the build fails on some systems.